validator/remote/src/main/java/tech/pegasys/teku/validator/remote/typedef/handlers/GetAggregateAttestationRequest.java

```diff
@@ -0,0 +1,126 @@
+ */
+
+package tech.pegasys.teku.validator.remote.typedef.handlers;
+
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.ATTESTATION_DATA_ROOT;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.COMMITTEE_INDEX;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.SLOT_QUERY_DESCRIPTION;
+import static tech.pegasys.teku.validator.remote.apiclient.ValidatorApiMethod.GET_AGGREGATE_V2;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.infrastructure.json.types.DeserializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.ssz.collections.SszBitvector;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.datastructures.operations.Attestation;
+import tech.pegasys.teku.spec.datastructures.operations.AttestationSchema;
+import tech.pegasys.teku.validator.remote.typedef.ResponseHandler;
+
+public class GetAggregateAttestationRequest extends AbstractTypeDefRequest {
+
+  final AttestationSchema<Attestation> attestationSchema;
+  final UInt64 slot;
+
+  private final ResponseHandler<GetAggregateAttestationResponse> responseHandler;
+
+  private final DeserializableTypeDefinition<GetAggregateAttestationResponse>
+      getAggregateAttestationTypeDef;
+
+  public GetAggregateAttestationRequest(
+      final HttpUrl baseEndpoint,
+      final OkHttpClient okHttpClient,
+      final Spec spec,
+      final UInt64 slot) {
+    super(baseEndpoint, okHttpClient);
+    this.slot = slot;
+    this.attestationSchema =
+        spec.atSlot(slot)
+            .getSchemaDefinitions()
+            .getAttestationSchema()
+            .castTypeToAttestationSchema();
+    this.getAggregateAttestationTypeDef =
+        DeserializableTypeDefinition.object(GetAggregateAttestationResponse.class)
+            .initializer(GetAggregateAttestationResponse::new)
+            .withField(
+                "version",
+                DeserializableTypeDefinition.enumOf(SpecMilestone.class),
+                GetAggregateAttestationResponse::getSpecMilestone,
+                GetAggregateAttestationResponse::setSpecMilestone)
+            .withField(
+                "data",
+                attestationSchema.getJsonTypeDefinition(),
+                GetAggregateAttestationResponse::getData,
+                GetAggregateAttestationResponse::setData)
+            .build();
+
+    this.responseHandler = new ResponseHandler<>(getAggregateAttestationTypeDef);
+  }
+
+  public Optional<Attestation> createAggregate(
+      final Bytes32 attestationHashTreeRoot, final UInt64 committeeIndex) {
+    final Map<String, String> queryParams = new HashMap<>();
+    queryParams.put(SLOT_QUERY_DESCRIPTION, slot.toString());
```

Comment:
`SLOT` not `SLOT_QUERY_DESCRIPTION`!

This should have been caught by a test...

Reply:
Fixed and added a test!